### ISSUE TITLE
Program lock status isn't remembered after a garbage collect

### ISSUE THREAD

**Issue by Aquateraneal** [mateoconlechuga/cesium#276]
I have multiple programs that I have created and locked on-calc. Whenever a garbage collect happens, then all of them unlock. This is very annoying, as I have to go back and manually re-lock them every time. The odd thing is, if I lock a program using TI-Connect CE, it stays locked after a garbage collect. TI-Connect CE is obviously doing something that Cesium isn't, would it be possible to implement whatever TI-Connect CE is doing into Cesium?

Steps to reproduce:
- Have two TI-Basic programs as examples. I will refer to these as `prgmCONLOCK` and `prgmCESLOCK`.
- Open `prgmCONLOCK` in TI-Connect CE and lock it, and lock `prgmCESLOCK` inside of Cesium.
- Run the `GarbageCollect` command.
  - *Note*: I am not actually running `GarbageCollect`, I am using TI-Boy CE, which sometimes needs a garbage collect after using. If need be, use TI-Boy CE for long enough that exiting causes a garbage collect, instead of running the `GarbageCollect` command directly.
- Now, `prgmCONLOCK` should still be locked, but `prgmCESLOCK` won't.

Any way to fix this?

**Comment by LogicalJoe** [comment]
Your programs are archived when you lock them.  Unarchive the program, lock it, and then rearchive it.

**Comment by Aquateraneal** [comment]
No, that's not what I'm saying. They lock fine, that's not the problem. I'm saying that, *after a GarbageCollect*, they lose their lock.
(Also, locking works fine when the programs are archived, I only have to un- and re-archive them when I'm hiding them... Unless you mean that it only gets remembered IF I un- and re-archive them, in which case why does NOTHING mention this?)
